base/namedtuple.jl

```diff
@@ -199,6 +200,7 @@ end
 if nameof(@__MODULE__) === :Base
     Tuple(nt::NamedTuple) = (nt...,)
     (::Type{T})(nt::NamedTuple) where {T <: Tuple} = (t = Tuple(nt); t isa T ? t : convert(T, t)::T)
+    convert(::Type{T}, nt::NamedTuple) where {T <: Tuple} = T(nt)
```

Comment:
This direction seems suspect to me, since it discards information. The other direction seems okay, since it adds information. Though also possibly misses a chance to error if you meant to have the right NamedTuple, and then happened to pass the wrong object.